src/ui/Analytics/Analytics.ts

```diff
@@ -289,32 +220,44 @@ export class Analytics extends Component {
    * This avoids logging every single partial query, which would make the reporting very confusing.
    *
    * @param actionCause Describes the cause of the event.
-   * @param meta The metadata which you can use to create custom dimensions. Can be an empty object ( `{}` ).
+   * @param meta The metadata which you want to use to create custom dimensions. Metadata can contain as many key-value
+   * pairs as you need. Each key must contain only alphanumeric characters and underscores. The Coveo Usage Analytics
+   * API automatically converts white spaces to underscores and uppercase characters to lowercase characters in key
+   * names. Each values must be a simple string. If you do not need to log metadata, you can simply pass an empty JSON
+   * ( `{}` ).
    */
   public logSearchAsYouType<T>(actionCause: IAnalyticsActionCause, meta: T) {
     this.client.logSearchAsYouType(actionCause, meta);
   }
 
   /**
-   * Logs a custom event on the service. You can use custom events to create custom reports, or to track events
+   * Logs a Custom event on the service. You can use custom events to create custom reports, or to track events
    * that are not queries or document views.
    *
    * @param actionCause Describes the cause of the event.
-   * @param meta The metadata which you can use to create custom dimensions. Can be an empty object ( `{}` ).
+   * @param meta The metadata which you want to use to create custom dimensions. Metadata can contain as many key-value
+   * pairs as you need. Each key must contain only alphanumeric characters and underscores. The Coveo Usage Analytics
+   * API automatically converts white spaces to underscores and uppercase characters to lowercase characters in key
+   * names. Each values must be a simple string. If you do not need to log metadata, you can simply pass an empty JSON
+   * ( `{}` ).
    * @param element The HTMLElement that the user has interacted with for this custom event.
    */
   public logCustomEvent<T>(actionCause: IAnalyticsActionCause, meta: T, element: HTMLElement = this.element) {
     this.client.logCustomEvent(actionCause, meta, element);
   }
 
   /**
-   * Logs a click event. You can understand click events as document views (e.g., clicking on a {@link ResultLink} or
+   * Logs a Click event. You can understand click events as document views (e.g., clicking on a {@link ResultLink} or
    * opening a {@link Quickview}).
    *
    * This event is logged immediately on the service.
    *
    * @param actionCause Describes the cause of the event.
-   * @param meta The metadata which you can use to create custom dimensions. Can be an empty object ( `{}` ).
+   * @param meta The metadata which you want to use to create custom dimensions. Metadata can contain as many key-value
```

Comment:
I will probably change this, depending on what the answer to this question is: https://discuss.coveo.com/t/logclickevent-metadata/1025